src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -31,6 +30,10 @@ public static partial class SymbolFinder
             return await FindOverridesArrayAsync(symbol, solution, projects, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <inheritdoc cref="FindOverridesAsync"/>
+        /// <remarks>
+        /// Use this overload to avoid boxing the result into an <see cref="IEnumerable{T}"/>.
+        /// </remarks>
         internal static async Task<ImmutableArray<ISymbol>> FindOverridesArrayAsync(
             ISymbol symbol, Solution solution, IImmutableSet<Project> projects = null, CancellationToken cancellationToken = default)
```

Comment:
the general pattern is that we have a public `IEnumerable<ISymbol> FindXXXAsync(...)` call, and an internal `ImmutableArray<ISymbol> FindXXXArrayAsync(...)`.  The latter is just for avoiding unnecessary boxing.  The former always delegates to the latter.